playwright-tests/e2e/tests/comments.spec.ts

```diff
@@ -0,0 +1,106 @@
+//comments-spec.ts
+import { expect } from "@playwright/test";
+import { test } from "../fixtures/index";
+import { Users } from "../../constants";
+import { TaskPage } from "../pom/tasks";
+import { Auth } from "../pom/login";
+import { faker } from "@faker-js/faker";
+import LoginPage from "../pom/login";
+
+test.describe("Comment Page", () => {
+  let todoName: string;
+  let assigneeComment: string;
+  let assignerComment: string;
+
+  test.beforeEach(async ({ page }) => {
+    todoName = faker.word.verb(5);
+    assigneeComment = faker.lorem.sentence();
+    assignerComment = faker.lorem.sentence();
+  });
+
+  test("comment on task as creator and assigned user", async ({
+    page,
+    loginPage,
+    browser,
+    taskPage,
+  }) => {
+    page.goto("/");
+
+    await test.step("step 1 - create a new task by assigner", async () => {
+      await taskPage.createTaskAndVerify({
+        taskName: todoName,
+        userName: "Sam Smith",
+      });
+    });
+
+    await test.step("step 2 - go to that newely created task", async () => {
+      await page.getByTestId("tasks-pending-table").getByText(todoName).click();
+
+      const endPoint =
+        todoName.split(" ").length > 1
+          ? todoName.split(" ").join("-")
+          : todoName;
+
+      await page.waitForResponse(response => response.url().includes(endPoint));
+      expect(await page.locator("h1").innerText()).toBe(todoName);
+    });
+
+    await test.step("step 3 - create comment as assigner in the todo and verify", async () => {
+      await taskPage.createCommentAndVerify({
+        comment: assignerComment,
+        taskName: todoName,
+      });
+    });
+
+    await test.step("step 4 - check of comment count was increased for the task", async () => {
+      await taskPage.checkForCount({
+        taskName: todoName,
+        count: 1,
+      });
+    });
+
+    const assigneeContext = await browser.newContext({
+      storageState: { cookies: [], origins: [] },
+    });
+
+    const assigneePage = await assigneeContext.newPage();
+    const assigneeAuth = new LoginPage(assigneePage);
+    const assigneeTasks = new TaskPage(assigneePage);
```

Comment:
if I move this to inside a step block it will effect the scoping right?  since I am using this in multiple steps I thought it is better to be outside of step block

Reply:
We can make use of `let` to share them among blocks.

Reply:
done